.github/workflows/build-frontend.yml

```
@@ -0,0 +1,67 @@
+name: Build Frontend
+run-name: Build frontend of ${{ inputs.frontend_name }} for ${{ inputs.deployment_name }} - ${{ github.event.inputs.ref }}
+
+on:
+  workflow_call:
+    inputs:
+      frontend_name:
+        description: 'Frontend type to build. Can be `frontend`, or `sr-frontend`'
+        type: string
+      repo_full_name:
+        description: 'Full repository name to build from.'
+        type: string
+        default: 'frzyc/genshin-optimizer'
+      ref:
+        description: 'Ref to build from. Can be a commit hash.'
+        type: string
+        default: 'master'
+      deployment_name:
+        description: 'Name for the deployment. This will determine the URL. If you choose a name that is already deployed, it will overwrite that deployment.'
+        type: string
+        default: 'master'
+      show_dev_components:
+        description: 'Flag to show components typically only shown in development mode, for experimental or in-progress features.'
+        type: boolean
+        default: false
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ inputs.repo_full_name }}
+          ref: ${{ inputs.ref }}
+          submodules: true
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'yarn'
+      - name: Yarn install
+        run: |
+          yarn install --immutable --immutable-cache
+      - name: Setup .env
+        run: |
+          printf '%s' "$ENVFILE" > apps/frontend/.env.local
+        env:
+          ENVFILE: ${{ secrets.ENVFILE }}
```

We would need to move the `ENVFILE` to vars
```suggestion
          ENVFILE: ${{ vars.ENVFILE }}
```

---

Good catch